src/main/java/io/github/bobocodebreskul/context/support/BeanDefinitionReaderUtils.java

```diff
@@ -60,35 +69,88 @@ public static String generateClassBeanName(Class<?> beanClass) {
-
-    Constructor<?>[] declaredConstructors = beanClass.getDeclaredConstructors();
-
-    validateBeanClassConstructorSize(beanClass.getName(), declaredConstructors);
+  public static List<BeanDependency> getConstructorBeanDependencies(
+      Constructor<?> beanConstructor) {
+    Objects.requireNonNull(beanConstructor, () -> {
+      log.error("Failed to get bean constructor dependencies for nullable constructor");
+      return "Bean constructor has not been specified";
+    });
+    log.trace("Scanning class {} for @Autowire candidates", beanConstructor.getDeclaringClass());
 
-    Stream<Class<?>> constructorDependenciesStream = Arrays.stream(declaredConstructors)
-        .flatMap(constructor -> Arrays.stream(constructor.getParameterTypes()));
+    return Arrays.stream(beanConstructor.getParameterTypes())
+        .map(dependency -> new BeanDependency(generateClassBeanName(dependency), dependency))
+        .toList();
+  }
 
-    Stream<Class<?>> fieldDependenciesStream = Arrays.stream(beanClass.getDeclaredFields())
-        .filter(field -> field.isAnnotationPresent(Autowired.class))
-        .map(Field::getType);
+  /**
+   * Finds and returns the initialization constructor for a given bean class. The initialization
+   * constructor is determined based on the presence of the {@link Autowired} annotation and the
+   * number of available constructors.
+   *
+   * @param beanClass the class of the bean to find the initialization constructor for.
+   * @param beanName  the name of the bean, used for logging and error handling.
+   * @return the initialization constructor for the specified bean class.
+   * @throws BeanDefinitionCreationException If there are issues with the bean's constructors, such
+   *                                         as having more than one constructor without the
+   *                                         {@link Autowired} annotation or multiple constructors
+   *                                         marked with {@link Autowired}.
+   */
+  public static Constructor<?> findBeanInitConstructor(Class<?> beanClass, String beanName) {
+    Constructor<?>[] declaredConstructors = beanClass.getDeclaredConstructors();
 
-    Stream<Class<?>> methodDependenciesStream = Arrays.stream(beanClass.getDeclaredMethods())
-        .filter(method -> method.isAnnotationPresent(Autowired.class))
-        .flatMap(method -> Arrays.stream(method.getParameterTypes()));
+    // One constructor with or without @Autowired.
+    if (declaredConstructors.length == 1) {
+      log.trace("One init constructor found and registered [{}] for bean candidate [{}]",
+          declaredConstructors[0], beanClass.getName());
+      return declaredConstructors[0];
+    } else if (declaredConstructors.length > 1) {
+      log.trace("Multiple constructors found for bean candidate [{}]", beanClass.getName());
+      if (isAnnotationPresentForAnyConstructor(Autowired.class, declaredConstructors)) {
```

Comment:
are we need this check? looks like we check the same in the next if, and as result lets just throw exception in the else block

Reply:
Yes, it's required to be here. This check verifies that we have at least one target annotation on our constructor, and next one verifies that only one constructor has it.